src/commands/cmd_cluster.cc

```diff
@@ -219,7 +210,7 @@ class CommandClusterX : public Commander {
         *output = redis::Error(s.Msg());
       }
     } else if (subcommand_ == "setslot") {
-      Status s = svr->cluster->SetSlot(slot_id_, args_[4], set_version_);
+      Status s = svr->cluster->SetSlot(slots_str_, args_[4], set_version_);
```

Comment:
Maybe, it would be better to parse slot ranges inside `CommandClusterX::Parse` and then pass already parsed (and valid) numbers to the `SetSlot` function.

Reply:
I'm not sure to do this. `CommandClusterX::Parse` is only used to find some keywords and do the simple check.
If necessary, I can migrate `parse` functions in `cluster.h` to `CommandClusterX`.

Reply:
I also think it would be better to do this in command parsing. Only valid slotranges is concerned in `cluster.cc` and these parsing aspects are not concerned.